src/components/SkyStateGenerator.cpp

```diff
@@ -183,12 +183,12 @@ const std::shared_ptr<SkyState> SkyStateGenerator::update(float dayRatio)
   return mCurrentSkyState;
 }
 
-std::tuple<unsigned int, unsigned int> SkyStateGenerator::findPresetIndices() const
+std::tuple<std::uint32_t, std::uint32_t> SkyStateGenerator::findPresetIndices() const
 {
-  unsigned int p1i = static_cast<unsigned int>(mSkyStatePresets.size()) - 1;
-  unsigned int p2i = 0;
+  std::uint32_t p1i = static_cast<std::uint32_t>(mSkyStatePresets.size()) - 1;
+  std::uint32_t p2i = 0;
 
-  for (unsigned int i = 0; i < mSkyStatePresets.size(); ++i)
+  for (std::uint32_t i = 0; i < mSkyStatePresets.size(); ++i)
```

Comment:
I think this needs a cast, otherwise there'll likely be a warning during compilation. For these simple casts, a C-Style cast is fine IMHO.

Reply:
I always look for compilation warnings with clang, and there was none. In fact I removed a bunch of compilation warnings ;) (except for one, where I added a todo instead). Should I still cast it?

Reply:
I wonder why that is. `mSkyStatePresets.size()` should return a number of type `size_t`, which is defined to be an integer large enough to hold the entire address space, ie. 64 bit on a 64 bit machine. That said, `uint32_t` is certainly not a 64-bit number.

Reply:
Hm I don't see the reason this should produce a warning. Both are unsigned and `i` is probably just casted to the wider type for the comparison?!  Which would result in no possible loss of information.

Reply:
Yeah this is really strange... Fixed now